### PR TITLE
sles4sap: Add support for SLES4SAP-15 (user settings, registration)

### PR DESCRIPTION
**Pull request by alvarocarvajald**
- Added user_settings for SLES4SAP-15 so that tests can be performed with an unprivileged user as in SLES.
- Removed installation/sles4sap_product_installation_mode for SLES4SAP-15.
- Changed order of SLES4SAP tests so patterns are tested first

### CODE REVIEW

File: products/sle/main.pm [hunk 1]
```diff
@@ -584,7 +584,7 @@ sub load_inst_tests {
         loadtest "installation/skip_registration" unless check_var('SLE_PRODUCT', 'leanos');
     }
     if (is_sles4sap) {
-        loadtest "installation/sles4sap_product_installation_mode";
+        loadtest "installation/sles4sap_product_installation_mode" unless sle_version_at_least('15');
```
Comment (okurz) [inline products/sle/main.pm]: then better combine the two checks into line 586?:

```perl
if (is_sles4sap && !sle_version_at_least('15')) {
```

**Comment by alvarocarvajald** [comment]
Defined SLES4SAP-15's default modules in lib/registration so the product is succesfully installed even in the presence of bsc#1056413, and added the SAP Applications' module to the array `@no_relnotes` in tests/installation/releasenotes.

Also created needles used by lib/registration in merge request: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/550

**Comment by okurz** [comment]
I hope you don't mind I made the subject line a bit more generic

**Comment by alvarocarvajald** [comment]
No problem. I should've updated the subject once I added more to the pull request. Thanks

**Pull request merged**